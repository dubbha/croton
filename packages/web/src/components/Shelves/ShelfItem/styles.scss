@import '../../../custom.scss';

.shelf-item {
  margin: 10px auto;
  display: flex;
  flex-direction: column;
  width: 30%;
  padding: 10px;
  min-height: 140px;

  &-new {
    display: flex;
    align-items: center;
    justify-content: center;
    min-height: 100%;
    min-width: 100%;
    color: $color-grass;

    &:hover {
      color: $color-grass-dark;
    }
  }
}
