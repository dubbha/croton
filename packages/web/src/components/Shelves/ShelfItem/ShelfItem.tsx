import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons';
import { Card } from 'react-bootstrap';

import '../styles.scss';

export type ShelfItemProps = {
  id: number,
  name: string,
  location: string,
  description: string,
  onSelect: (id: number) => void
}

export const ShelfItem = ({
  id,
  name,
  location,
  description,
  onSelect,
} :ShelfItemProps) => {
  return (
    <div className="shelf-item">
      <Card text="primary" onClick={() => onSelect(id)}>
        <Card.Img variant="top" src="" />
        <Card.Body>
          <Card.Title>{name}</Card.Title>
          <Card.Subtitle className="mb-2 text-muted">
            <FontAwesomeIcon icon={faMapMarkerAlt} />
            <span className="icon-prefix">{location}</span>
          </Card.Subtitle>
          <Card.Text>
            {description}
          </Card.Text>
        </Card.Body>
      </Card>
    </div>
  );
};
