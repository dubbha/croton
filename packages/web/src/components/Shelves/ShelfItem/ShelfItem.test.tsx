import React from 'react';
import { render } from '@testing-library/react';
import { ShelfItem } from './ShelfItem';

describe('components/ShelfItem', () => {
  const props = {
    id: 12,
    name: 'name',
    location: 'location',
    description: 'description',
    onSelect: () => {},
  };

  it('should render successfully', () => {
    const { container } = render(<ShelfItem {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
