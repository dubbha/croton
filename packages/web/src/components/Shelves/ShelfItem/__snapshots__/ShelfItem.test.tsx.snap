// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ShelfItem should render successfully 1`] = `
<div
  class="shelf-item"
>
  <div
    class="card text-primary"
  >
    <img
      class="card-img-top"
      src=""
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      />
      <div
        class="mb-2 text-muted card-subtitle h6"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-map-marker-alt fa-w-12 "
          data-icon="map-marker-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            fill="currentColor"
          />
        </svg>
        <span
          class="icon-prefix"
        />
      </div>
      <p
        class="card-text"
      />
    </div>
  </div>
</div>
`;
